test/jdk/java/awt/dnd/RejectDragDropActionTest.java

```diff
@@ -95,15 +95,16 @@ public static void main(String[] args) throws Exception {
             startPoint.translate(50, 50);
             endPoint.translate(150, 150);
 
-            robot.delay(DELAY_TIME);
-
             robot.mouseMove(startPoint.x, startPoint.y);
             robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
             for (Point p = new Point(startPoint); !p.equals(endPoint);
                  p.translate(sign(endPoint.x - p.x),
                          sign(endPoint.y - p.y))) {
                 robot.mouseMove(p.x, p.y);
-                robot.delay(50);
+                if (incorrectActionDetected) {
+                    System.err.println("Incorrect action during loop");
+                    break;
+                }
```

Comment:
Actually, I meant adding it as part of the condition in the for-loop itself. However, it works either way.

The initial bug report for [JDK-4774532](https://bugs.openjdk.org/browse/JDK-4774532) says `dragOver` shouldn't be called at all. Do we want to update the test for a stricter assertion. For me, `dragOver` is never called.

Reply:
It does look cleaner as a part of the for-loop itself, so I moved it in. For updating the test with a stricter assertion, maybe it's better to leave this as is for now since it doesn't necessarily hurt the test from what I can tell.